mlem/contrib/docker/base.py

```diff
@@ -571,10 +571,7 @@ def build_image(self, context_dir: str) -> DockerImage:
         with self.env.daemon.client() as client:
             if self.push:
                 self.image.registry.login(client)
-
-            if self.force_overwrite:
-                self.image.delete(client)  # to avoid spawning dangling images
-            elif self.image.exists(client):
+            if not self.force_overwrite and self.image.exists(client):
```

Comment:
@mike0sv, did this to adhere to default Docker behavior. Got the previous behavior as unexpected: couple of times I interrupted `mlem build docker`, but was expecting the image to still be here. Was surprised to see it gone. I guess we can add a special param in future, like `--cleanup_if_dangling` or something.